### ISSUE TITLE
HPX 1.7.0-rc1 fails to build icw APEX + OTF2

### ISSUE THREAD

**Issue by kordejong** [STEllAR-GROUP/hpx#5395]
## Expected Behavior

Build succeeds.

## Actual Behavior

Build fails with these messages (full log attached):

```
apex-src/src/apex/otf2_listener.cpp:1402:59: error: ‘gather_there’ is not a member of ‘hpx::lcos’; did you mean ‘hpx::collectives::gather_there’?
apex-src/src/apex/otf2_listener.cpp:1412:28: error: ‘gather_here’ is not a member of ‘hpx::lcos’; did you mean ‘hpx::collectives::gather_here’?
apex-src/src/apex/otf2_listener.cpp:1455:59: error: ‘gather_there’ is not a member of ‘hpx::lcos’; did you mean ‘hpx::collectives::gather_there’? apex-src/src/apex/otf2_listener.cpp:1464:32: error: ‘gather_here’ is not a member of ‘hpx::lcos’; did you mean ‘hpx::collectives::gather_here’?
apex-src/src/apex/otf2_listener.cpp:1526:32: error: ‘broadcast_from’ is not a member of ‘hpx::lcos’; did you mean ‘broadcast_apply’?
apex-src/src/apex/otf2_listener.cpp:1526:58: error: expected primary-expression before ‘>’ token               
apex-src/src/apex/otf2_listener.cpp:1531:32: error: ‘broadcast_to’ is not a member of ‘hpx::lcos’; did you mean ‘broadcast’?
apex-src/src/apex/otf2_listener.cpp:1585:59: error: ‘gather_there’ is not a member of ‘hpx::lcos’; did you mean ‘hpx::collectives::gather_there’?
apex-src/src/apex/otf2_listener.cpp:1594:32: error: ‘gather_here’ is not a member of ‘hpx::lcos’; did you mean ‘hpx::collectives::gather_here’?
apex-src/src/apex/otf2_listener.cpp:1662:32: error: ‘broadcast_from’ is not a member of ‘hpx::lcos’; did you mean ‘broadcast_apply’?
apex-src/src/apex/otf2_listener.cpp:1662:58: error: expected primary-expression before ‘>’ token               
apex-src/src/apex/otf2_listener.cpp:1667:32: error: ‘broadcast_to’ is not a member of ‘hpx::lcos’; did you mean ‘broadcast’?
apex-src/src/apex/otf2_listener.cpp:1714:59: error: ‘gather_there’ is not a member of ‘hpx::lcos’; did you mean ‘hpx::collectives::gather_there’?
apex-src/src/apex/otf2_listener.cpp:1724:28: error: ‘gather_here’ is not a member of ‘hpx::lcos’; did you mean ‘hpx::collectives::gather_here’?
```

## Steps to Reproduce the Problem

  1. Configure HPX with `-DHPX_WITH_APEX:BOOL=ON -DAPEX_WITH_OTF2:BOOL=ON -DHELLO_HPX_BUILD_OTF2:BOOL=ON -DHPX_WITH_EXAMPLES:BOOL=ON`
  
I test HPX builds using this project: https://github.com/kordejong/hello_hpx The build that fails is the one in https://github.com/kordejong/hello_hpx/blob/b8f53591c9b6fcfcc6e5ce887bd48c23ffe2c77d/rel_with_deb_info_build.sh#L12

## Specifications

  - HPX Version: 1.7.0-rc1
  - Platform (compiler, OS): gcc-9.3.0, Linux

[build_log.txt](https://github.com/STEllAR-GROUP/hpx/files/6662989/build_log.txt)

**Comment by hkaiser** [comment]
@kordejong Apparently the collective functions are used more than I expected. We have introduced a breaking change to those functions to unify their API and straighten out naming, namespaces and argument handling. The following changes have been made:

- All collective operations now can be found in `namespace hpx::collectives`
- The sequence of arguments has been unified for all collective operations
- In order to make it harder for users to get the sequence of arguments wrong (many arguments were simply of type `size_t`) we have introduced helper types that ensure that the arguments are unambiguous (see `num_sites_arg`, `this_site_arg`, etc.)

We have updated the documentation to reflect those changes: https://hpx-docs.stellar-group.org/branches/master/html/libs/full/collectives/api.html#modules-collectives-api.

I apologize for assuming breaking APIs wouldn't go noticed. I will try to go back to this and re-add the old APIs for backwards compatibility purposes.

**Comment by msimberg** [comment]
Note that if the breaking changes don't affect your code directly but only through APEX this has already been fixed in APEX. The fix will be included in 1.7.0-rc2. We've also added a CI configuration with OTF2 enabled to catch this in the future.

**Comment by kordejong** [comment]
My little `hello_hpx` project only builds HPX. It does not actually use HPX. I use it to test whether I can still build HPX as I am used to. I will test 1.7.0-rc2 once it is released and will then also build my own project. I am pretty sure my own project does not use the API calls mentioned in the above error message.
